_drafts/oryx/ssr.md

```diff
@@ -0,0 +1,132 @@
+
+Moreover, Oryx employs a "late hydration" strategy, delaying the hydration process until the user interacts with a component. This ensures that client-side resources are only utilized when absolutely necessary, fostering an efficient and highly responsive user experience.
+
+
+## Developing with SSR
+
+### SSR Consideration
+
+When developing with SSR, it's important to understand how the SSR process works. SSR involves rendering the initial HTML on the server, which is then sent to the client. This provides faster initial page load times and better SEO. However, because this process can differ from traditional client-side rendering, there are specific considerations and potential pitfalls to keep in mind, such as avoiding direct manipulation of the DOM and being mindful of lifecycle hooks.
+
+### Configuration
+
+Oryx integrates seamlessly with server environments, offering support for both traditional Node.js server-side rendering (SSR) and serverless architectures, akin to AWS Lambda or Netlify functions.
+
+For Node.js SSR, Oryx leverages ExpressJS, a minimalist web framework for Node.js. By utilizing the `createServer` method, Oryx is capable of spinning up an ExpressJS server configured for SSR out of the box.
+
+Simultaneously, Oryx caters to the increasing demand for serverless architectures. The `storefrontHandler` method enables SSR in AWS Lambda-like environments, providing a flexible solution for serverless deployments.
+
+While both approaches offer sensible configuration tailored to most deployment scenarios, it also exposes a lower-level API, allowing advanced customization to cater to unique project requirements.
+
+### Building with SSR Support
+
+Designed with SSR at its core, Oryx ensures that all components correctly render server-side. This design choice brings about inherent advantages such as rapid initial page load times and boosted SEO, right out of the box.
+
+Additionally, Oryx features mechanisms that further enhance performance by managing the hydration process intelligently.
+
+
+### SSR-aware Components
+
+Oryx components are built with SSR in mind. They're designed to render correctly on the server and work with late and partial hydration. However, when building custom components, be aware of the SSR lifecycle and the potential issues that may arise, such as the `unsafeStatic` issue, which can lead to unexpected behavior during the SSR process.
+
+### Decorators
+
+Oryx provides special decorators to address some SSR related challenges:
+
+- The `@hydratable` decorator marks a component for late hydration. This allows the component to render on the server but delays its hydration until it's interacted with on the client. Hydration can be triggered programmatically, or automatically with events.   
+
+- The `@ssrShim` decorator shims some component API in order to make it work on the Server: `toggleAttribute()` and `style` property. 
```

Comment:
Is there a better way to construct this sentence? I find it hard to connect the two parts

Reply:
fixed